### PR TITLE
x86_64: reuse cmp lhs mcval if it is already in a register

### PR DESCRIPTION
**Pull request by micahswitzer**
I'm fairly confident this change is sound, but I'm not super familiar with the x86_64 backened so I might have missed something.

### CODE REVIEW

File: src/arch/x86_64/CodeGen.zig [hunk 1]
```diff
@@ -5483,15 +5483,16 @@ fn airCmp(self: *Self, inst: Air.Inst.Index, op: math.CompareOperator) !void {
...
+                const tmp_reg = try self.copyToTmpRegister(ty, lhs);
+                break :blk self.register_manager.lockRegAssumeUnused(tmp_reg);
+            },
         };
-        defer if (lhs_lock) |lock| self.register_manager.unlockReg(lock);
-
-        const dst_reg = try self.copyToTmpRegister(ty, lhs);
-        const dst_reg_lock = self.register_manager.lockRegAssumeUnused(dst_reg);
         defer self.register_manager.unlockReg(dst_reg_lock);
+        const dst_reg = dst_reg_lock.register;
```
Comment (micahswitzer) [inline src/arch/x86_64/CodeGen.zig:5495]: Pulling the register out of the lock feels a bit goofy to me, but it was the cleanest way I could come up with. I'd love suggestions for improvements.

**Comment by kubkon** [comment]
Can you enable a matching behaviour test that this change fixes?

**Comment by micahswitzer** [comment]
> Can you enable a matching behaviour test that this change fixes?

This is not a bugfix, it's just a small optimization. Sorry for not making that very clear.

**Comment by micahswitzer** [comment]
Superseded by #15140

**Pull request closed without merging**